changelog.d/20732_datadog_agent_zstd.enhancement.md

```diff
@@ -0,0 +1 @@
+Allow the datadog agent source to accept payloads that have been compressed with `zstd`.
```

Comment:
```suggestion
Allow the `datadog_agent` source to accept payloads that have been compressed with `zstd`.
```

I think we've standardized on either the component name, in backticks, or using capitalization like "Datadog Agent source".